### DISCUSSION TITLE
Not able to display image from web API which returns byte array - .NET Maui Android application

### DISCUSSION THREAD

**Discussion by mach-9** [dotnet/maui#13364] [General]
I'm new to mobile development with .NET Maui, and trying to display a picture (.jpg) from an API that I created.

I'm having issues with trying to display the byte array returned from my web API on a .NET Maui page. I've followed the Community Toolkit Maui sample project (ByteArrayToImageSourceConverterPage), and only changed the client call to my web API, see code below:

Community Toolkit Maui sample code
DotNetBotImageByteArray = await client.GetByteArrayAsync("https://user-images.githubusercontent.com/13558917/137551073-ac8958bf-83e3-4ae3-8623-4db6dce49d02.png", cancellationTokenSource.Token).ConfigureAwait(false);

To:
DotNetBotImageByteArray = await client.GetByteArrayAsync("http://XXX.XXX.XXX/api/[controller name]/[Method]/{[value]}", cancellationTokenSource.Token).ConfigureAwait(false);

The web API returns the byte array: byte[] results = _IService.GetImageAsync([value]);

I've tried to change the API to return ToBase64String and Maui to FromBase64String, too many other options to list, googled the heck out of it, and having no success.

Another option would be to FileWrite the bytes from the API to a folder on the phone, and then display the photo from the folder. Doesn't make sense to me that I should have to do that, all I need is the byte array from the API, store it in a variable, and bind it to the image.

I'm currently using the toolkit sample project so I haven't changed anything, just added my call to the API instead.

Using VS2022, and developing an android application using a simulator.

Any help on this would be much appreciated.

Thank you.

**Comment by carlos-leitek** [comment]
facing exactly the same situation. loading file from Web
Api in Maui projecty Android dont work. but browsers like chrome can open the image link and show image.
Have tryed different things . My WebAPI server is linux. My MAUI project is Android tested.
Any news from help ? Im using server with dns xxx.ddns.net with code: 

  <Image x:Name="img" Grid.Row="1"
        HeightRequest="250"
        WidthRequest="150"
        HorizontalOptions="FillAndExpand"
        Aspect="AspectFit"
        Source="https://xxxx.xxxx.pt/content/uploads/20230226_141307-Altran-0-24-1.png" >                     
        </Image>

**Reply by mach-9** [reply]
Hi Carlos,

No other comments or help on this.

My preference would not be to write the image to a folder in IIS and then call it from the mobile app., but if I have to make it work, then I have to.

Good luck.

**Comment by cat0363** [comment]
Hi, @mach-9 
What about converting the image data stored as a Byte array to an ImageSource via a memory stream?

      byte[] imgData = _IService.GetImageAsync([value]);
      ImageSource imgSource;
      imgSource = ImageSource.FromStream(() => new MemoryStream(imgData));

Simply bind or set the above ImageSource to the Image control's Source property.

**Reply by darutter0508** [reply]
I'm having a similar issue. In my case the call to get the byte array never returns when called from the Android version of the MAUI app. The same code works when run against an iOS device (iPhone). Is there a setting in the Android environment that needs to be set??

**Reply by mach-9** [reply]
Darutter,

Instead of using .NET Maui for android development, I switched to Android Studio using Java. I found it very challenging trying to get stuff to work in .NET Maui, much easier using Android Studio. Good luck.
